barretenberg/cpp/src/barretenberg/common/parallel_for_mutex_pool.cpp

```diff
@@ -124,10 +125,16 @@ namespace bb {
 void parallel_for_mutex_pool(size_t num_iterations, const std::function<void(size_t)>& func)
 {
     static ThreadPool pool(get_num_cpus() - 1);
-
+    static std::atomic_bool nested = false;
```

Comment:
note that if this is used safely, we don't need the std::atomic, but if we are catching the mess up case of nesting parallel_for this should be atomic

Reply:
gonna just make that a comment